docs_src/microservices/security/Ch-SSH-Tunneling-HowToSecureDeviceServices.md

```diff
@@ -0,0 +1,185 @@
+# Security for EdgeX Stack
+
+This page describes one of options to secure the EdgeX software stack with running remote device services like device-virtual, device-rest, device-mqtt, and so on, via secure two-way SSH-tunnelings.
+
+## Basic SSH-Tunneling
+
+In this option to secure the EdgeX software stack, SSH tunneling is utilized. The basic idea is to create a secure SSH connection between  a local machine, the primary host, and a remote machine, the secondary, in which some micro-services or applications can be relayed.   In this particular example, the primary host is  running the whole EdgeX core services including core services and security services but **without** any device service.  The device services are running in the secondary or the remote machine.  
+
+The communication is secure because SSH port forwarding connection is encrypted by default.
+
+The SSH communication is established by introducing some extra SSH-related services:
+
+1) device-ssh-proxy: this is the service with ssh client opening up the SSH communication between the primary and the secondary
+
+2) device-ssh-remote: this is actually the SSH server or daemon service together with device services running on the remote machine
+
+The high-level diagram is shown as follows:
+
+![image](ssh-tunneling device.png) "Top level diagram for SSH tunneling for device services"
+
+<TBD> more description needed
+
+## How to- Reference implementation example
+
+### Setup remote running Virtual Machine
+
+In the example setup, `vagrant` is used on the top of `Virtual Box` to set up as the secondary/remote VM. The network port for ssh standard port 22 is mapped into the vm port 2223 and host port also 2223.
```

Comment:
I think one of these ports should be 2222 and the other 2223?

Reply:
yea, good catch.  2222 is used internally by ` vagrant` vm itself.  I also created another mapping rule to map 2223:2223 for vm container to host. so there are two mappings here: 2222:22 by vagrant itself and 2223:2223 for vagrant vm -> host.  The sshd container itself is mapped with `-p 2223:22`.